web/modules/custom/hoeringsportal_citizen_proposal/src/Form/ProposalApproveForm.php

```diff
@@ -0,0 +1,145 @@
+    return 'proposal_approve_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): RedirectResponse|array {
+    $adminFormStateValues = $this->state->get('citizen_proposal_admin_form_values');
+
+    if (!$entity = $this->helper->tempStoreValid()) {
+      return $this->helper->abandonSubmission();
+    }
+
+    $form['approve_form_help'] = [
+      '#type' => 'processed_text',
+      '#format' => $adminFormStateValues['approve_intro']['format'] ?? 'filtered_html',
+      '#text' => $adminFormStateValues['approve_intro']['value'] ?? '',
+    ];
+
+    $form['approve_form_title'] = [
+      '#prefix' => '<h5>' . $this->t('Title') . '</h5>',
+      '#type' => 'processed_text',
+      '#text' => $entity?->title->value ?? '',
+      '#format' => 'filtered_html',
+    ];
+
+    $form['approve_form_proposal'] = [
+      '#prefix' => '<h5>' . $this->t('Proposal') . '</h5>',
+      '#type' => 'processed_text',
+      '#text' => $entity?->field_proposal->value ?? '',
+      '#format' => 'filtered_html',
+    ];
+
+    $form['approve_form_remarks'] = [
+      '#prefix' => '<h5>' . $this->t('Remarks') . '</h5>',
+      '#type' => 'processed_text',
+      '#text' => $entity?->field_remarks->value ?? '',
+      '#format' => 'filtered_html',
+    ];
+
+    $form['approve_form_name'] = [
+      '#prefix' => '<h5>' . $this->t('Name') . '</h5>',
+      '#type' => 'processed_text',
+      '#text' => $entity?->field_author_name->value ?? '',
+      '#format' => 'filtered_html',
+    ];
+
+    $form['approve_form_email'] = [
+      '#prefix' => '<h5>' . $this->t('E-mail') . '</h5>',
+      '#type' => 'processed_text',
+      '#text' => $entity?->field_author_email->value ?? '',
+      '#format' => 'filtered_html',
+    ];
+
+    $form['actions']['#type'] = 'actions';
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Approve proposal'),
+      '#button_type' => 'primary',
+    ];
+
+    $form['actions']['edit_link'] = [
+      '#title' => $this->t('Edit proposal'),
+      '#type' => 'link',
+      '#url' => Url::fromRoute('hoeringsportal_citizen_proposal.citizen_proposal.proposal_add'),
+      '#attributes' => ['class' => ['btn', 'btn-secondary', 'mb-2']],
+    ];
+
+    $form['actions']['cancel'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Cancel proposal'),
+      '#button_type' => 'link',
+      '#submit' => [[$this, 'cancelSubmit']],
+      '#attributes' => ['class' => ['btn', 'btn-link']],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    if (!$entity = $this->helper->tempStoreValid()) {
+      return $this->helper->abandonSubmission();
+    }
+
+    $this->messenger()->addStatus('Thank you for you submission.');
+    $entity->save();
+    $this->helper->tempStoreDelete();
+    $form_state
+      ->setRedirect('<front>');
```

Comment:
Maybe we should consider adding a custom “Thanks for your proposal” page where we can show a message from the `ProposalAdminForm` settings rather than just sending the citizen to the front page.

Reply:
That can be implemented in a future PR